src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final ColorSupplier COLOR_SUPPLIER = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Type figures = Type.values()[new Random().nextInt(Type.values().length)];
+        switch (figures) {
+            case SQUARE: {
+                Square square = new Square(RANDOM.nextInt(100),
+                        COLOR_SUPPLIER.getRandomColor().toString().toLowerCase());
+                return square;
+            }
+            case ISOSCELESTRAPEZOID: {
+                IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid(RANDOM.nextInt(100),
+                        RANDOM.nextInt(100), RANDOM.nextInt(100),
+                        COLOR_SUPPLIER.getRandomColor().toString().toLowerCase());
+                return isoscelesTrapezoid;
+            }
+            case CIRCLE: {
+                Circle circle = new Circle(RANDOM.nextInt(100),
+                        COLOR_SUPPLIER.getRandomColor().toString().toLowerCase());
+                return circle;
+            }
+            case RESTANGLE: {
+                Rectangle rectangle = new Rectangle(RANDOM.nextInt(100), RANDOM.nextInt(100),
+                        COLOR_SUPPLIER.getRandomColor().toString().toLowerCase());
+                return rectangle;
+            }
+            case RIGHTTRIANGLE: {
+                RightTriangle rightTriangle = new RightTriangle(RANDOM.nextInt(100),
+                        RANDOM.nextInt(100),
+                        COLOR_SUPPLIER.getRandomColor().toString().toLowerCase());
+                return rightTriangle;
+            }
+            default: {
+                return null;
```

Comment:
throw IllegalStateException, always better to be upfront about the problem rather than return null silently. This state shouldn't occur - each figure has to have a case for it